nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-processors/src/main/java/org/apache/nifi/processors/iceberg/converter/GenericDataConverters.java

```diff
@@ -47,45 +51,106 @@
  */
 public class GenericDataConverters {
 
-    static class SameTypeConverter extends DataConverter<Object, Object> {
+    static class PrimitiveTypeConverter extends DataConverter<Object, Object> {
+        final Type.PrimitiveType targetType;
+        final DataType sourceType;
+
+        public PrimitiveTypeConverter(final Type.PrimitiveType type, final DataType dataType) {
+            targetType = type;
+            sourceType = dataType;
+        }
 
         @Override
         public Object convert(Object data) {
-            return data;
+            switch (targetType.typeId()) {
+                case BOOLEAN:
+                    return DataTypeUtils.convertType(data, RecordFieldType.BOOLEAN.getDataType(), null);
+                case INTEGER:
+                    return DataTypeUtils.convertType(data, RecordFieldType.INT.getDataType(), null);
+                case LONG:
+                    return DataTypeUtils.convertType(data, RecordFieldType.LONG.getDataType(), null);
+                case FLOAT:
+                    return DataTypeUtils.convertType(data, RecordFieldType.FLOAT.getDataType(), null);
+                case DOUBLE:
+                    return DataTypeUtils.convertType(data, RecordFieldType.DOUBLE.getDataType(), null);
+                case DATE:
+                    return DataTypeUtils.toLocalDate(data, () -> StringUtils.isEmpty(sourceType.getFormat())
+                            ? null
+                            : DataTypeUtils.getDateTimeFormatter(sourceType.getFormat(), ZoneId.systemDefault()), null);
+                case UUID:
+                    return DataTypeUtils.convertType(data, RecordFieldType.UUID.getDataType(), null);
+                case STRING:
+                default:
+                    return DataTypeUtils.convertRecordFieldtoObject(data, RecordFieldType.STRING.getDataType());
+            }
         }
     }
 
-    static class TimeConverter extends DataConverter<Time, LocalTime> {
+    static class TimeConverter extends DataConverter<Object, LocalTime> {
+
+        private final String timeFormat;
+
+        public TimeConverter(final String format) {
+            this.timeFormat = format;
+        }
 
         @Override
-        public LocalTime convert(Time data) {
-            return data.toLocalTime();
+        public LocalTime convert(Object data) {
+            if (data instanceof Time) {
```

Comment:
This type check is unnecessary because `DataTypeUtils.toTime ` contains the same logic and `toLocalTime` is also  called in the below line.